GpsDataAnalyzer/fileparser/fileparser.py

```diff
@@ -46,14 +50,115 @@ def parse_file(self, filename) -> GpsDataSet:
         file_type = self.get_file_type(filename)
         
         if file_type == 'xml':
+            logger.info('Parse xml file ' + filename)
             return self.parse_xml(filename)
         elif file_type == 'csv':
+            logger.info('Parse csv file ' + filename)
             return self.parse_csv(filename)
         else:
-            print("Invalid file type!")
+            logger.debug('Invalid file type.')
             return None
 
 
+    def parse_time(self, timestr) -> datetime:
+        """
+        Parse the xml timestamp and return a new datetime object
+
+        Args:
+          timestr: string of the timestamp extracted from xml file
+
+        Returns:
+          A datetime object with timezone
+        """
+        #Replace timezone Z with '' since python datetime parser could not handler %Z
+        timestr = timestr.replace('Z', '', 1)
+        #Parse time in the format "%Y-%m-%dT%H:%M:%S.%f"
+        time = datetime.strptime(timestr, "%Y-%m-%dT%H:%M:%S.%f")
+        #Set timezone back
+        tz_time = time.replace(tzinfo=timezone.utc)
```

Comment:
probably no need to create tz_time?

just: return time.replace(tzinfo=timezone.utc)